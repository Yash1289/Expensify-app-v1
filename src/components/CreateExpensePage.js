const React = require("react");
import ExpenseForm from "./ExpenseForm"
import { startExpenseAdder } from "../actions/expenses"
import { connect } from "react-redux"

export class CreateExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.CreateExpense(expense)
        this.props.history.push("/")
    }

    render(){
        return (
            <div>
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title"> Add Expense</h1>
                </div>
             </div>
             <div className="content-container"> 
                <ExpenseForm onSubmit={this.onSubmit} />
             </div>  
            </div>
        )
    }
}


const mapDispatchToProps = (dispatch) => (
    {
        CreateExpense: (expense) => dispatch(startExpenseAdder(expense))
    }
)
export default connect(undefined , mapDispatchToProps)(CreateExpensePage)