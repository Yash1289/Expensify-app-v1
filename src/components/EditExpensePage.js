const React = require("react");
import { connect } from "react-redux"
import ExpenseForm from "./ExpenseForm"
import { startExpenseEdit , startExpenseRemover } from "../actions/expenses"

export class EditExpensePage extends React.Component { 
    onSubmit = (expense) => {
        this.props.ExpenseEdit(this.props.expense.id , expense)
        this.props.history.push("/")
    }
    onClick = () => {
        console.log(this.props.expense.id)
        this.props.ExpenseRemover({ id: this.props.expense.id })
        this.props.history.push("/")
    }
    render() {
        return (
            <div>
                <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
                <button onClick={this.onClick} >Remove</button>
            </div>
        )
    }
}

const mapStateToProps = (state , props ) => {
    return {
        expense : state.expenses.find((expense) => expense.id === props.match.params.id )
    }
}

const mapDispatchToProps = ( dispatch , props ) => {
    return {
        ExpenseEdit: (id , expense) => dispatch(startExpenseEdit(id , expense)),
        ExpenseRemover : (data) => dispatch(startExpenseRemover(data))
    }
}

export default connect(mapStateToProps , mapDispatchToProps)(EditExpensePage) 