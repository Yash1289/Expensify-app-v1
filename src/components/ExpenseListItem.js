import React from "react";
import {expenseRemover} from "../actions/expenses"
import { NavLink } from "react-router-dom"
import { connect } from "react-redux";
import moment from "moment"
const rupeeFormatter = require("rupee-formatter")

const ExpenseListItem = ({  id , description , amount, createdAt}) => (
    <div>
        <NavLink to={`/edit/${id}`} ><h3>{description}</h3></NavLink>
        <p>
        {rupeeFormatter(amount)} - {moment(createdAt).format("MMMM DD , YYYY")}
        </p>
    </div>
)



export default ExpenseListItem ;