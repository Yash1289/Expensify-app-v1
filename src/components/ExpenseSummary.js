import React from 'react'
import filterExpenses from "../selectors/expenses"
import { connect } from "react-redux"
import selectExpensesTotal from "../selectors/ExpensesTotal"

export const ExpensesSummary = ({ expensesCount , expensesTotal }) => (
    <div>
        <h1>Viewing {expensesCount} {expensesCount === 1 ? " expense" : " expenses"} totalling {expensesTotal}</h1>
    </div>
    
)

const mapStateToProps = (state) => {
    const filteredExpenses = filterExpenses(state.expenses, state.filter)
    return {
        expensesCount: filteredExpenses.length ,
        expensesTotal : selectExpensesTotal(filteredExpenses)
    }
}

const ConnectedExpensesSummary = connect(mapStateToProps)(ExpensesSummary)

export default ConnectedExpensesSummary