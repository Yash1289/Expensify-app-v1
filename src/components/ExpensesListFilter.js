import React from "react"
import { connect } from "react-redux"
import { setTextFilter , sortByDate , sortByAmount , setStartDate , setEndDate } from "../actions/filter"
import { DateRangePicker } from "react-dates"

export class ExpensesListFilter extends React.Component{
    state = {
        calendarFocused : null
    }
    onDatesChange = ({ startDate , endDate}) => {
        this.props.SetStartDate(startDate)
        this.props.SetEndDate(endDate)
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    }
    onTextChange = (e) => {
        this.props.SetTextFilter(e.target.value)
    }
    onSortByChange = (e) => {
        console.log(e.target.value)
        if (e.target.value === "date") {
            this.props.SortByDate()
        }
        else if (e.target.value === "amount") {
            this.props.SortByAmount()
        }
    }
   render(){
       return (
           <div className="content-container">
                <div className="input-group">
                   <div className="input-group__item hello">
                       <input className="text-input" type="text" placeholder="Search Expenses" value={this.props.filter.text} onChange={this.onTextChange} />
                   </div>
                   <div className="input-group__item">
                       <select className="select" value={this.props.filter.sortBy} onChange={this.onSortByChange}>
                           <option value="date">Date</option>
                           <option value="amount">Amount</option>
                       </select>
                   </div>
                   <div className="input-group__item">
                       <DateRangePicker
                           startDate={this.props.filter.startDate}
                           endDate={this.props.filter.endDate}
                           onDatesChange={this.onDatesChange}
                           focusedInput={this.state.calendarFocused}
                           onFocusChange={this.onFocusChange}
                           showClearDates={true}
                           numberOfMonths={1}
                           isOutsideRange={() => false}
                       />
                   </div>
                </div> 
           </div>
       )
   }
}

const mapStateToProps = (state) => {
    return{
        filter : state.filter
    }
}

const mapDispatchToProps = (dispatch , props) => {
    return {
        SetStartDate : (startDate) => dispatch(setStartDate(startDate)),
        SetEndDate : (endDate) => dispatch(setEndDate(endDate)),
        SetTextFilter : (text) => dispatch(setTextFilter(text)) , 
        SortByDate : () => dispatch(sortByDate()) ,
        SortByAmount : () => dispatch(sortByAmount())
    }
}

export default connect(mapStateToProps , mapDispatchToProps)(ExpensesListFilter)