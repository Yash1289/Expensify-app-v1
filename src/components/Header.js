const React = require("react");
import { Link , NavLink} from "react-router-dom"

const Header = () => (
    <div>
        <h1>Expensify App</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName = "is-active">Create</NavLink>
    </div>
)

export default Header;