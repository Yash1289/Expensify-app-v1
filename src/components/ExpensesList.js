import React from 'react'
import ExpenseListItem from "./ExpenseListItem"
import filterExpenses from "../selectors/expenses"
import { connect } from "react-redux"



export const ExpensesList = (props) => (
    <div>
        {props.expenses.length === 0 && <p>No Expenses</p>}
        {
            props.expenses.map((expense) => <ExpenseListItem key={expense.id} {...expense} />)
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: filterExpenses(state.expenses, state.filter)
    }
}

const ConnectedExpensesList = connect(mapStateToProps)(ExpensesList)

export default ConnectedExpensesList