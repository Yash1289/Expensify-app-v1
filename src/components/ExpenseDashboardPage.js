const React = require("react");
import ConnectedExpensesList from  "./ExpensesList"
import ExpensesListFilter from  "./ExpensesListFilter"
import ExpensesSummary from "./ExpenseSummary"

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpensesListFilter />
        <ConnectedExpensesList />
    </div>
)

export default ExpenseDashboardPage ;