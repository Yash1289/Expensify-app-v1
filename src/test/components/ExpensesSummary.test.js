import React from "react"
import { shallow } from "enzyme"
import {ExpensesSummary} from "../../components/ExpenseSummary"
import { TestScheduler } from "jest"

test("should correctly render ExpensesSummary with 1 expense", () => {
    const wrapper  = shallow(<ExpensesSummary expensesCount = {1} expensesTotal = {256.35} />)
    expect(wrapper).toMatchSnapshot()
})

test("Should correctly render ExpensesSummary with 2 expenses" ,() => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={3526.3}/>)
    expect(wrapper).toMatchSnapshot()
})