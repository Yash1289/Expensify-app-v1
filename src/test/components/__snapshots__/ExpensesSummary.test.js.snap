// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpensesSummary with 2 expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     
     expenses
     totalling 
    3526.3
  </p>
</div>
`;

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
     expense
     totalling 
    256.35
  </p>
</div>
`;
